lib/ui/channel_buffers.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart.ui;
+
+/// A saved platform message for a channel with its callback.
+class StoredMessage {
+  final ByteData _data;
+  final PlatformMessageResponseCallback _callback;
+
+  StoredMessage(this._data, this._callback);
+  ByteData get data => _data;
```

Comment:
I think we should drop these (`data` and `callback` getters) and just make the object opaque for now. We don't want anyone mutating these things out from under us if they somehow get a hold of them, and we only need to use them internally right?  

That may necessitate some other dart:ui method for sending a `StoredMessage`, but I think it's preferable to having the framework have access to these members.

Reply:
Interesting thought but I don't think it's necessary.  We are using the getters for testing.  The only thing that is mutable is the ByteData, if you really wanted to control what gets put there you can always just send a message to that channel with whatever data you want, that's public API.

Reply:
Do we need to test these things though? I'd be more worried about someone trying to use this to modify the ByteData on us, which is not immutable.